napari/utils/perf/_timers.py

```diff
@@ -192,14 +224,21 @@ def block_timer(
         print(f"{name} {event.duration_ms:.3f}ms")
 
 
-def _create_timer():
+def _create_timer() -> Tuple[PerfTimers, Callable, Callable, Callable]:
```

Comment:
Did we know anything about arguments and return type of callables?


Fast check suggest that it should something like:

```
Callable[[str, Optional[str], bool, ...], contextlib.AbstractContextManager],
```
and then reflect signatures of `add_instant_event` and `add_counter_event` 

With such improved type annotation, mypy could check if proper arguments are passed to returned functions.

Reply:
These are tricky, if not impossible to type because of the `**kwargs` in all the funciton signatures. See https://docs.python.org/3/library/typing.html#annotating-callables:

>Callable cannot express complex signatures such as functions that take a variadic number of arguments, [overloaded functions](https://docs.python.org/3/library/typing.html#overload), or functions that have keyword-only parameters.

I also tried using `Protocol` classes to do this, but couldn't work out how to without mypy erroring.

I think having `Callable` here is still an improvement over no typing though?